metals/src/main/scala/scala/meta/internal/metals/NewFilesProvider.scala

```diff
@@ -66,31 +135,80 @@ class NewFilesProvider(
       )
   }
 
+  private def createWorksheet(directory: Option[AbsolutePath], name: String)(
+      implicit ec: ExecutionContext
+  ): Future[AbsolutePath] = {
+    val path = directory.getOrElse(workspace).resolve(name + ".worksheet.sc")
+    createFile(path)
+  }
+
   private def createFile(
       path: AbsolutePath
-  )(implicit ec: ExecutionContext): Future[URI] = {
+  )(implicit ec: ExecutionContext): Future[AbsolutePath] = {
     val result = Future {
-      Files.createFile(path.toNIO).toUri()
+      AbsolutePath(
+        Files.createFile(path.toNIO)
```

Comment:
You can use `touch` from `XtensionAbsolutePathBuffers`

Reply:
`touch` does not return the path, and although I already have the path, I like to reuse output of the function.

Reply:
Can we change touch to return the path?